internal/engine/upper.go

```diff
@@ -290,7 +290,7 @@ func handleBuildStarted(ctx context.Context, state *store.EngineState, action Bu
 	state.CurrentlyBuilding = mn
 }
 
-func handleCompletedBuild(ctx context.Context, engineState *store.EngineState, cb BuildCompleteAction) error {
+func (u *Upper) handleCompletedBuild(ctx context.Context, engineState *store.EngineState, cb BuildCompleteAction) error {
```

Comment:
hmmmm...i don't think this needs to be a method of Upper anymore?

i'm going to imminently change the API so that the reducers aren't allowed to have access to any state, so making this a method will be impossible soon